http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1291,4 +1328,30 @@ public RequestDistribution(HttpClient httpClient, Processor processor)
             });
         }
     }
+
+    private static class SecureHttpClient
+            extends HttpClient
+    {
+        private final JettyAuthenticationStore jettyAuthenticationStore;
+
+        public SecureHttpClient(HttpClientTransport transport, SslContextFactory sslContextFactory)
+        {
+            super(transport, sslContextFactory);
+            jettyAuthenticationStore = new JettyAuthenticationStore();
```

Comment:
The findAuthentication method for HttpAuthenticationStore can be very slow as it iterates through a list. We need a faster lookup as discussed with Dain.